sstables/sstables.cc

```diff
@@ -320,28 +320,28 @@ inline void write(sstable_version_types v, file_writer& out, const utils::UUID&
 // are contiguous, it is not always the case. So we want to have the
 // flexibility of parsing them separately.
 template <typename Size>
-future<> parse(sstable_version_types v, random_access_reader& in, disk_string<Size>& s) {
+future<> parse(schema_ptr schema, sstable_version_types v, random_access_reader& in, disk_string<Size>& s) {
     auto len = std::make_unique<Size>();
-    auto f = parse(v, in, *len);
-    return f.then([v, &in, &s, len = std::move(len)] {
-        return parse(v, in, *len, s.value);
+    auto f = parse(schema, v, in, *len);
+    return f.then([v, schema, &in, &s, len = std::move(len)] {
```

Comment:
I think it is fine to use schema=std::move(schema) here, and std::move(schema) below for, again, a bit less reference counting?